src/devices_models/devices/common/add_constraint_dual.jl

```diff
@@ -57,7 +57,7 @@ function assign_dual_variable!(
     container::OptimizationContainer,
     constraint_type::Type{<:ConstraintType},
     service::D,
-    ::AbstractServiceFormulation,
+    ::Type{AbstractServiceFormulation},
```

Comment:
@jd-lara Should this be `::Type{<:AbstractServiceFormulation}`?

Reply:
Yes